skrub/datasets/_utils.py

```diff
@@ -1,8 +1,38 @@
-import os
 from pathlib import Path
 
 
-def get_data_dir(name: str | None = None) -> Path:
+def get_data_home(data_home: Path | None = None) -> Path:
+    """Returns the path of the skrub data directory.
+
+    This folder is used by some large dataset loaders to avoid downloading the
+    data several times.
+
+    By default the data directory is set to a folder named 'skrub_data' in the
+    user home folder.
+
+    Alternatively, it can be set programmatically by giving an explicit folder
+    path. The '~' symbol is expanded to the user home folder.
+
+    If the folder does not already exist, it is automatically created.
+
+    Parameters
+    ----------
+    data_home : Path, default=None
```

Comment:
As a user, I think that I would like to pass a `str` as well. If passing an `str`, then we can convert it to a `Path` internally as a first step:

```python
if isinstance(data_home, str):
    data_home = Path(data_home)
```